### PR TITLE
PUBDEV-7586 - Documentation changes related to the official K8S Docker image

### PR DESCRIPTION
**Pull request by Pscheidl**
https://0xdata.atlassian.net/browse/PUBDEV-7586

### CODE REVIEW

File: h2o-k8s/README.md [hunk 1]
```diff
@@ -100,6 +100,9 @@ The name of the application `app: h2o-k8s` must correspond to the name expected
 for the H2O node discovery to work. H2O communicates on port 54321, therefore `containerPort: 54321`must be exposed to
 make it possible for the clients to connect.
 
+There are official H2O Docker images available in Docker hub: `h2oai/h2o-open-source-k8s:<tagname>`. Please make sure to replace the `<tagname>`
```
Comment (michalkurka) [inline h2o-k8s/README.md:103]: I would even just say `latest` in the example... and leave a note here about nightly. WDYT?

Comment (michalkurka) [reply h2o-k8s/README.md:103]: It is nice when examples are good to go for the 99.999% use cases.

Comment (Pscheidl) [reply h2o-k8s/README.md:103]: Done. Also added RAM usage note from Docker hub, as this is important.

Comment (angela0xdata) [reply h2o-k8s/README.md:103]: can you specify a specific release like rel-zahradnik/4?

Comment (Pscheidl) [reply h2o-k8s/README.md]: Yes, this is possible. It'll be the `PROJECT_VERSION`, e.g. `3.30.0.5`

Comment (Pscheidl) [reply h2o-k8s/README.md]: There are none like that ATM, it's too new.

File: h2o-docs/src/product/welcome.rst [hunk 2]
```diff
@@ -1003,11 +1003,20 @@ We strongly recommended to run H2O as a StatefulSet on a Kubernetes cluster. Tre
...
-- ``H2O_NODE_LOOKUP_TIMEOUT`` - (Optional) The node lookup constraint. Specify the time before the node lookup times out. (Defaults to 3 minutes.)
-- ``H2O_NODE_EXPECTED_COUNT`` - (Optional) The node lookup constraint. This is the expected number of H2O pods to be discovered. (Defaults to 3.)
+- ``H2O_KUBERNETES_SERVICE_DNS`` - **[MANDATORY]** Crucial for the clustering to work. The format usually follows the ``<service-name>.<project-name>.svc.cluster.local`` pattern. This setting enables H2O node discovery via DNS. It must be modified to match the name of the headless service created. Also, pay attention to the rest of the address. It must match the specifics of your Kubernetes implementation.
+- ``H2O_NODE_LOOKUP_TIMEOUT`` - **[OPTIONAL]** Node lookup constraint. Specify the time before the node lookup times out.
+- ``H2O_NODE_EXPECTED_COUNT`` - **[OPTIONAL]** Node lookup constraint. This is the expected number of H2O pods to be discovered.
+- ``H2O_KUBERNETES_API_PORT`` - **[OPTIONAL]** Port for Kubernetes API checks to listen on. Defaults to 8080.
+
+If none of the optional lookup constraints is specified, a sensible default node lookup timeout will be set - currently
```
Comment (hannah-tillman) [inline h2o-docs/src/product/welcome.rst:1013]: "If none of the optional lookup constraints **is** specified" --minor grammar change; should be "are"

Comment (Pscheidl) [inline h2o-docs/src/product/welcome.rst:1013]:
```suggestion
If none of the optional lookup constraints are specified, a sensible default node lookup timeout will be set - currently
```

Comment (Pscheidl) [reply h2o-docs/src/product/welcome.rst]: Thank you. Corrected.

**Pull request merged**